drivers/i2c/i2c_tca9546a.c

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) 2020 Innoseis BV
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_tca9546a
+
+#include <zephyr.h>
+#include <device.h>
+#include <devicetree.h>
+#include <drivers/i2c.h>
+#include <logging/log.h>
+#include <stdint.h>
+
+LOG_MODULE_REGISTER(tca9546a, CONFIG_I2C_LOG_LEVEL);
+
+struct tca9546a_root_config {
+	const struct device *bus;
+	uint16_t slave_addr;
+};
+
+struct tca9546a_root_data {
+	struct k_mutex lock;
+	uint8_t select_reg;
+#ifdef CONFIG_PM_DEVICE
+	uint32_t power_state;
+#endif
+};
+
+struct tca9546a_channel_config {
+	const struct device *root;
+	uint8_t addr;
```

Comment:
Does it's an I2C address or the channel index ?

Reply:
That's the channel index, I will rename it

Reply:
ok I see, in this case could be great to rename also tca9546a_set_selection to tca9546a_set_chan/tca9546a_set_channel, and select_reg to select_chan
see https://github.com/torvalds/linux/blob/master/drivers/i2c/muxes/i2c-mux-pca954x.c

Reply:
Actually it's more of a mask inside the address register